packages/yarnpkg-fslib/sources/path.ts

```diff
@@ -137,11 +137,11 @@ function fromPortablePath(p: Path): NativePath {
 
 // Path should look like "N:/berry/scripts/plugin-pack.js"
 // And transform to "/N:/berry/scripts/plugin-pack.js"
-function toPortablePath(p: Path): PortablePath {
+function toPortablePath(p: Path | Buffer): PortablePath {
   if (process.platform !== `win32`)
     return p as PortablePath;
 
-  p = p.replace(/\\/g, `/`);
+  p = p.toString().replace(/\\/g, `/`);
```

Comment:
Forcing it to a string will hide issues like https://github.com/yarnpkg/berry/issues/1889 so this should at least be behind a `Buffer.isBuffer` check.

Reply:
Good idea. Fixed.